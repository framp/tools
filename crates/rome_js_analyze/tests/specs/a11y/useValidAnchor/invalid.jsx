<>
	<a />
	<a href />
	<a href={null} />
	<a href={undefined} />
	<a href="" />
	<a href="#" />
	<a href={"#"} />
	<a href={`#`} />
	<a href="javascript:void(0)" />
	<a href={"javascript:void(0)"} />
	<a href={`javascript:void(0)`} />
	<a href={<span className="token string">javascript:void(0)</span>} />
	<a onClick={() => void 0} />
	<a href="#" onClick={() => void 0} />
	<a href="javascript:void(0)" onClick={() => void 0} />
	<a href={"javascript:void(0)"} onClick={() => void 0} />
</>;
