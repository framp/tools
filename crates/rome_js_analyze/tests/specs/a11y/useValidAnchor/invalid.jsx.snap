---
source: crates/rome_js_analyze/tests/spec_tests.rs
expression: invalid.jsx
---
# Input
```js
<>
	<a />
	<a href />
	<a href={null} />
	<a href={undefined} />
	<a href="" />
	<a href="#" />
	<a href={"#"} />
	<a href={`#`} />
	<a href="javascript:void(0)" />
	<a href={"javascript:void(0)"} />
	<a href={`javascript:void(0)`} />
	<a href={<span className="token string">javascript:void(0)</span>} />
	<a onClick={() => void 0} />
	<a href="#" onClick={() => void 0} />
	<a href="javascript:void(0)" onClick={() => void 0} />
	<a href={"javascript:void(0)"} onClick={() => void 0} />
</>;

```

# Diagnostics
```
invalid.jsx:2:2 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Provide a href attribute for the a element.
  
    1 │ <>
  > 2 │ 	<a />
      │ 	^^^^^
    3 │ 	<a href />
    4 │ 	<a href={null} />
  
  i An anchor element should always have a href
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:3:5 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! The attribute href has to be assigned to a valid value.
  
    1 │ <>
    2 │ 	<a />
  > 3 │ 	<a href />
      │ 	   ^^^^^
    4 │ 	<a href={null} />
    5 │ 	<a href={undefined} />
  
  i The href attribute should be a valid a URL
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:4:11 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Provide a valid value for the attribute href.
  
    2 │ 	<a />
    3 │ 	<a href />
  > 4 │ 	<a href={null} />
      │ 	         ^^^^
    5 │ 	<a href={undefined} />
    6 │ 	<a href="" />
  
  i The href attribute should be a valid a URL
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:5:11 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Provide a valid value for the attribute href.
  
    3 │ 	<a href />
    4 │ 	<a href={null} />
  > 5 │ 	<a href={undefined} />
      │ 	         ^^^^^^^^^
    6 │ 	<a href="" />
    7 │ 	<a href="#" />
  
  i The href attribute should be a valid a URL
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:7:10 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Provide a valid value for the attribute href.
  
    5 │ 	<a href={undefined} />
    6 │ 	<a href="" />
  > 7 │ 	<a href="#" />
      │ 	        ^^^
    8 │ 	<a href={"#"} />
    9 │ 	<a href={`#`} />
  
  i The href attribute should be a valid a URL
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:8:11 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Provide a valid value for the attribute href.
  
     6 │ 	<a href="" />
     7 │ 	<a href="#" />
   > 8 │ 	<a href={"#"} />
       │ 	         ^^^
     9 │ 	<a href={`#`} />
    10 │ 	<a href="javascript:void(0)" />
  
  i The href attribute should be a valid a URL
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:9:12 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Provide a valid value for the attribute href.
  
     7 │ 	<a href="#" />
     8 │ 	<a href={"#"} />
   > 9 │ 	<a href={`#`} />
       │ 	          ^
    10 │ 	<a href="javascript:void(0)" />
    11 │ 	<a href={"javascript:void(0)"} />
  
  i The href attribute should be a valid a URL
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:10:10 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Provide a valid value for the attribute href.
  
     8 │ 	<a href={"#"} />
     9 │ 	<a href={`#`} />
  > 10 │ 	<a href="javascript:void(0)" />
       │ 	        ^^^^^^^^^^^^^^^^^^^^
    11 │ 	<a href={"javascript:void(0)"} />
    12 │ 	<a href={`javascript:void(0)`} />
  
  i The href attribute should be a valid a URL
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:12:12 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Provide a valid value for the attribute href.
  
    10 │ 	<a href="javascript:void(0)" />
    11 │ 	<a href={"javascript:void(0)"} />
  > 12 │ 	<a href={`javascript:void(0)`} />
       │ 	          ^^^^^^^^^^^^^^^^^^
    13 │ 	<a href={<span className="token string">javascript:void(0)</span>} />
    14 │ 	<a onClick={() => void 0} />
  
  i The href attribute should be a valid a URL
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:13:11 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Provide a valid value for the attribute href.
  
    11 │ 	<a href={"javascript:void(0)"} />
    12 │ 	<a href={`javascript:void(0)`} />
  > 13 │ 	<a href={<span className="token string">javascript:void(0)</span>} />
       │ 	         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    14 │ 	<a onClick={() => void 0} />
    15 │ 	<a href="#" onClick={() => void 0} />
  
  i The href attribute should be a valid a URL
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:14:5 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Use a button element instead of an a element.
  
    12 │ 	<a href={`javascript:void(0)`} />
    13 │ 	<a href={<span className="token string">javascript:void(0)</span>} />
  > 14 │ 	<a onClick={() => void 0} />
       │ 	   ^^^^^^^^^^^^^^^^^^^^^^
    15 │ 	<a href="#" onClick={() => void 0} />
    16 │ 	<a href="javascript:void(0)" onClick={() => void 0} />
  
  i Anchor elements should only be used for default sections or page navigation
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:15:2 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Use a button element instead of an a element.
  
    13 │ 	<a href={<span className="token string">javascript:void(0)</span>} />
    14 │ 	<a onClick={() => void 0} />
  > 15 │ 	<a href="#" onClick={() => void 0} />
       │ 	^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    16 │ 	<a href="javascript:void(0)" onClick={() => void 0} />
    17 │ 	<a href={"javascript:void(0)"} onClick={() => void 0} />
  
  i Anchor elements should only be used for default sections or page navigation
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:16:2 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Use a button element instead of an a element.
  
    14 │ 	<a onClick={() => void 0} />
    15 │ 	<a href="#" onClick={() => void 0} />
  > 16 │ 	<a href="javascript:void(0)" onClick={() => void 0} />
       │ 	^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    17 │ 	<a href={"javascript:void(0)"} onClick={() => void 0} />
    18 │ </>;
  
  i Anchor elements should only be used for default sections or page navigation
  
  i Check this thorough explanation to better understand the context.
  

```

```
invalid.jsx:17:2 lint/a11y/useValidAnchor ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Use a button element instead of an a element.
  
    15 │ 	<a href="#" onClick={() => void 0} />
    16 │ 	<a href="javascript:void(0)" onClick={() => void 0} />
  > 17 │ 	<a href={"javascript:void(0)"} onClick={() => void 0} />
       │ 	^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    18 │ </>;
    19 │ 
  
  i Anchor elements should only be used for default sections or page navigation
  
  i Check this thorough explanation to better understand the context.
  

```


